### PR TITLE
plugins/inventory/cobbler: Collect IP addresses for hosts and add opt…

### PR DESCRIPTION
**Pull request by opoplawski**
…ion to collect all DNS name to IP address mappings

##### SUMMARY
It can be very helpful to have easy access to a host's primary IP address and various DNS name to IP address mappings defined in cobbler.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/inventory/cobbler.py

##### ADDITIONAL INFORMATION
```paste below
- debug:
    msg: "My IP address is {{ ip_address }}"
- debug:
    msg: "The IP address of server.example.com is {{ hostvars['server.example.com']['ip_address'] }}"
- debug:
    mesg "The IP address of server-interface.example.com is {{ cobbler_ip_addresses['server-interface.example.com'] }}"
```

### CODE REVIEW

File: plugins/inventory/cobbler.py [hunk 1]
```diff
@@ -80,6 +80,11 @@
         description: Toggle, if V(true) the plugin will retrieve host facts from the server
         type: boolean
         default: true
+      want_ip_addresses:
+        description: Toggle, if V(true) the plugin will add a C(cobbler_ip_addresses) and C(cobbleer_ipv6_addresses) dictionary to the defined O(group) mapping interface DNS names to IP addresses
```
Comment (felixfontein) [inline plugins/inventory/cobbler.py]:
```suggestion
        description: Toggle, if V(true) the plugin will add a C(cobbler_ip_addresses) and C(cobbleer_ipv6_addresses) dictionary to the defined O(group) mapping interface DNS names to IP addresses.
```

File: plugins/inventory/cobbler.py [hunk 2]
```diff
@@ -296,8 +303,45 @@ def parse(self, inventory, loader, path, cache=True):
                 self.inventory.add_child(self.group, hostname)
 
             # Add host variables
+            ip_address = None
+            ipv6_address = None
+            for (iname, ivalue) in iteritems(interfaces):
```
Comment (felixfontein) [inline plugins/inventory/cobbler.py:310]: While `iteritems()` is a bit more efficient with Python 2.7, using `items()` is IMO better readable:
```suggestion
            for iname, ivalue in interfaces.items():
```

File: changelogs/fragments/6711-cobbler-ip-address.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - cobbler inventory plugin - add ``want_ip_addresses`` option to collect all interface DNS name to IP address mapping (https://github.com/ansible-collections/community.general/pull/6711).
+  - cobbler inventory plugin - add primay IP addess to ``ip_address`` and IPv6 address to ``ipv6_address`` host variable (https://github.com/ansible-collections/community.general/pull/6711)
```
Comment (felixfontein) [inline changelogs/fragments/6711-cobbler-ip-address.yml:3]:
```suggestion
  - cobbler inventory plugin - add primay IP addess to ``ip_address`` and IPv6 address to ``ipv6_address`` host variable (https://github.com/ansible-collections/community.general/pull/6711).
```

File: plugins/inventory/cobbler.py [hunk 4]
```diff
@@ -66,20 +66,27 @@
...
         default: cobbler_
       want_facts:
-        description: Toggle, if V(true) the plugin will retrieve host facts from the server
+        description: Toggle, if V(true) the plugin will retrieve host facts from the server.
         type: boolean
         default: true
+      want_ip_addresses:
+        description:
+          - Toggle, if V(true) the plugin will add a C(cobbler_ip_addresses) and C(cobbleer_ipv6_addresses) dictionary to the defined O(group) mapping
+          - interface DNS names to IP addresses.
```
Comment (felixfontein) [inline plugins/inventory/cobbler.py:86]:
```suggestion
            interface DNS names to IP addresses.
```

**Comment by opoplawski** [comment]
One question I have is whether or not it is okay to simply add the `ip_address` and `ipv6_address` variables to the host variables.  Not sure if there is any standard among inventory plugins.

**Comment by felixfontein** [comment]
One problem with such generic names is that you can likely interfere with existing uses of these names in existing playbooks, roles, or inventories. Usually it's better to prefix them with something that's specific to this inventory plugin; for example, the community.docker.docker_containers inventory plugin prefixes everything with `docker_`. (This of course can still cause conflicts, but it is a little bit less likely :) )

**Comment by opoplawski** [comment]
Yeah, I'm going to think about it for a little bit, but I think that a `cobbler_` prefix probably is in order.

**Comment by opoplawski** [comment]
Though I think `cobbler_ip_address` *might* be confused with the IP address of the cobbler server.

File: plugins/inventory/cobbler.py [hunk 5]
```diff
@@ -296,8 +306,45 @@ def parse(self, inventory, loader, path, cache=True):
...
+
             if self.get_option('want_facts'):
                 try:
                     self.inventory.set_variable(hostname, 'cobbler', host)
                 except ValueError as e:
                     self.display.warning("Could not set host info for %s: %s" % (hostname, to_text(e)))
+
+        if self.get_option('want_ip_addresses'):
+            self.inventory.set_variable(self.group, 'cobbler_ipv4_addresses', ip_addresses)
+            self.inventory.set_variable(self.group, 'cobbler_ipv6_addresses', ipv6_addresses)
```
Comment (felixfontein) [inline plugins/inventory/cobbler.py:350]: `cobbler_ipv4_addresses` and `cobbler_ipv6_addresses` are neither mentioned in the changelog nor in the documentation.

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this by tomorrow morning.

**Comment by felixfontein** [comment]
@opoplawski thanks for your contribution!

**Pull request merged**